.github/workflows/ci-test.yml

```diff
@@ -0,0 +1,43 @@
+name: CI Experiment
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: "master"
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Setup Maven
+        env:
+          ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+          ARTIFACTORY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+        run: |
+          mkdir -p ~/.m2
+          cat <<EOF > ~/.m2/settings.xml
+          <settings>
+            <servers>
+              <server>
+                <id>artifactory-snapshots</id>
+                <username>${ARTIFACTORY_USERNAME}</username>
+                <password>${ARTIFACTORY_PASSWORD}</password>
+              </server>
+              <server>
+                <id>artifactory-releases</id>
+                <username>${ARTIFACTORY_USERNAME}</username>
+                <password>${ARTIFACTORY_PASSWORD}</password>
+             </server>
+           </servers>
+          </settings>
+          EOF
+      - name: Build with Maven
+        run: |
+          mvn verify -P it-cassandra-4.0
```

Comment:
should we use the wrapper please? and verify is doing the tests as well, was this the intention?

best would be imo:

* job 1 `Test` - ./mvnw test
* job 2 `Integration test C4` -  `/mvnw integration-test -P it-cassandra-4.0`
* job 3 `Integration test C3` -  `/mvnw integration-test`
* job 4 `Integration test DSE` -  `/mvnw integration-test -P dse`

First job can be prerequisite for the other three, which can run in parallel.. This will be the best, as you run unit tests only once and then only int tests for each profile..

Reply:
@tatu-at-datastax: I like the approach @ivansenic recommends above, especially since it avoids an issue with the current GCB workflows in that each run executes the full suite of unit tests (i.e. all 3 persistence modules are tested in each of the 3 workflows). 

If our overall goal is to reduce runtime, we could even consider running all 4 jobs listed above in parallel? We're generally pretty good at checking in code that works :).

Reply:
@jeffreyscarpenter you did not watch my _I code, I pollute_ presentation :smile: We should always confirm unit tests are green before firing the int tests.. This way we save unnecessary work if the those fail.. And those should be fast, under 2 mins for example..

Reply:
But there is more to this, we cal always test docker image building as well for example..